File: C++/3dprinter.cpp
/*
 *	Problem: 3D Printed Statues
 */
 
#include <iostream>
#include <string>
#include <sstream>
#include <cmath>

using namespace std;

/*-----main-----*/
int main() {
	std::string line;
	getline(cin, line);

	int numStatues;
	std::istringstream(line) >> numStatues;

	double numDays = log2(numStatues);	
	int minNumDays = ceil(numDays);
	minNumDays += 1;
		
	std::cout << minNumDays;
	return 0;
}

File: Ruby/3dprinter.rb
=begin
 *
 *	Problem:		3D Printed Statues
 *	Problem ID:	3dprinter
 *	url:				https://open.kattis.com/problems/3dprinter
 *	Solution:		Ruby
 *
=end

numStatues = gets.to_i

numDays = ( Math.log(numStatues, 10) / Math.log(2, 10) )
minNumDays = numDays.ceil
minNumDays += 1

puts (minNumDays)